eth2near/eth2near-block-relay-rs/src/eth2near_relay.rs

```diff
@@ -122,78 +137,19 @@ impl Eth2NearRelay {
-                        .as_bytes(),
-                );
-
-                if self.eth_client_contract.is_known_block(&hash)? {
-                    return Ok(true);
-                } else {
-                    trace!(target: "relay", "Block with slot={} not found on Near", slot);
-                    return Ok(false);
-                }
-            }
-            Err(err) => {
-                trace!(target: "relay", "Error \"{}\" in getting beacon block body for slot={}", err, slot)
-            }
-        }
-
-        Ok(false)
-    }
-
-    fn search_slot_forward(&self, slot: u64) -> Result<u64, Box<dyn Error>> {
-        let mut current_step = 1;
-        while self.block_known_on_near(slot + current_step)? {
-            current_step *= 2;
-        }
-
-        self.search_slot_backward(slot, slot + current_step)
-    }
-
-    fn search_slot_backward(&self, start_slot: u64, last_slot: u64) -> Result<u64, Box<dyn Error>> {
-        let mut start_slot = start_slot;
-        let mut last_slot = last_slot;
-        while start_slot + 1 < last_slot {
-            let mid_slot = start_slot + (last_slot - start_slot) / 2;
-            if self.block_known_on_near(mid_slot)? {
-                start_slot = mid_slot;
-            } else {
-                last_slot = mid_slot;
-            }
-        }
-
-        Ok(start_slot)
-    }
-
-    fn get_last_slot(&mut self) -> Result<u64, Box<dyn Error>> {
-        trace!(target: "relay", "= Search for last slot on near =");
-
-        let mut slot = self.eth_client_contract.get_last_submitted_slot();
-
-        let finalized_slot = self.eth_client_contract.get_finalized_beacon_block_slot()?;
-        trace!(target: "relay", "Finalized slot on near={}", finalized_slot);
-
-        slot = max(finalized_slot, slot);
-        trace!(target: "relay", "Init slot for search as {}", slot);
-
-        if self.block_known_on_near(slot)? {
-            self.search_slot_forward(slot)
-        } else {
-            self.search_slot_backward(finalized_slot, slot)
-        }
+    // get the slot numbers between the last submitted slot and signature slot for next update
+    // if we sending updates once in 'update_submission_frequency' epochs
+    // `update_submission_frequency * ONE_EPOCH_IN_SLOTS` -- gap in slots between two finalized
+    //  blocks in neighboring updates.
+    // `2 * ONE_EPOCH_IN_SLOTS` -- gap between finalized and attested block.
+    // `1` -- expected gap between attested block slot and signature slot
+    fn get_gap_between_finalized_and_signature_slot(update_submission_frequency: u64) -> u64 {
+        const EXPECTED_EPOCHS_BETWEEN_HEAD_AND_FINALIZED_BLOCKS: u64 = 2;
+        const EXPECTED_SLOTS_BETWEEN_ATTESTED_AND_SIGNATURE_SLOTS: u64 = 1;
+
+        update_submission_frequency * ONE_EPOCH_IN_SLOTS +
+            EXPECTED_EPOCHS_BETWEEN_HEAD_AND_FINALIZED_BLOCKS * ONE_EPOCH_IN_SLOTS +
+            EXPECTED_SLOTS_BETWEEN_ATTESTED_AND_SIGNATURE_SLOTS
```

Comment:
```suggestion
        update_submission_frequency * ONE_EPOCH_IN_SLOTS
            + EXPECTED_EPOCHS_BETWEEN_HEAD_AND_FINALIZED_BLOCKS * ONE_EPOCH_IN_SLOTS
            + EXPECTED_SLOTS_BETWEEN_ATTESTED_AND_SIGNATURE_SLOTS
```

Reply:
Soon I will learn to run `cargo fmt` before commit :) https://github.com/aurora-is-near/rainbow-bridge/pull/775/commits/c3ea734bcfad8f1f568df67fcf6fe89128fdfb36